### DISCUSSION TITLE
Windows 11 taskbar context menus

### DISCUSSION THREAD

**Discussion by pljones** [jamulussoftware/jamulus#3074] [UI Related]
The Windows 11 taskbar can allow applications to display custom right-click context menus.  Not all do, however:

| | |
|--------|--------|
| Snipping Tool | Thunderbird |
| ![image](https://github.com/jamulussoftware/jamulus/assets/1549463/f029c8fa-5106-4c35-854c-d1fce168baf1) | ![image](https://github.com/jamulussoftware/jamulus/assets/1549463/89f2f477-1520-4fb9-8306-50d2c676926d) |
| Jamulus | Reaper | 
| ![image](https://github.com/jamulussoftware/jamulus/assets/1549463/3fb2ca15-9ec4-4ca8-82fd-cbfb9bd80eb6) | ![image](https://github.com/jamulussoftware/jamulus/assets/1549463/a9c3fe51-ee1c-4c63-a0b5-96756195de7d) |

Now, Reaper previously had "Recent Projects" on its taskbar context menu.  They've disappeared with Windows 11, I think.  So something changed.
I'd like to know:
a) How do Snipping Tool and Thunderbird add items to the menu?
b) Does Reaper still _try_ to do it but fail?  If so, why?
c) **Could Jamulus offer "Recent connections"?**

Not being either a GUI developer or a Windows developer, I'm a bit lost as to where to start looking into this.  So any help would be appreciated.

----

Possibly relevant links:
* https://stackoverflow.com/questions/38035921/how-to-add-an-entry-to-toolbar-context-menu-in-qt

**Comment by rdica** [comment]
@pljones On a side note, I was looking for something like this in Ubuntu Studio 22.04 after noticing that Carla taskbar icon right click shows a history of which config files are loaded by Carla. The Jamulus taskbar icon didnt have any kind of history for config or connections either.

**Reply by henkdegroot** [reply]
It seems that part of the control of this menu is build into the windows functionality. It works based upon recent opened items. 
I am guessing, when you double click a reaper file in the windows explorer which then opens reaper will make that item appear in the recent item list unless you have switch off the Jump Lists function in your windows settings. See https://www.thewindowsclub.com/fix-jump-list-missing-disappeared-permanently-windows-7?utm_content=cmp-true

As the recent connection is not a opened file, I think this can't be made into a list of items in this menu.

I do see other applications having options for Tasks, so most likely a task to initiate a connection could be an option.

**Reply by henkdegroot** [reply]
Here a link to another project which has implemented some tasks:
https://github.com/microsoft/terminal/issues/576

And a QT reference here; 
https://doc.qt.io/qt-5/qml-qtwinextras-jumplist.html

**Reply by pljones** [reply]
Hm, I also noticed on my newly-updated other now-Win11 box, Reaper still has the pinned projects...  Just I see _no_ way to add extras...  Talk about useability...
